Bricks/template/is_unique_ptr.h

```diff
@@ -56,6 +56,41 @@ static_assert(std::is_same<int, typename is_unique_ptr<int>::underlying_type>::v
 static_assert(is_unique_ptr<std::unique_ptr<int>>::value, "");
 static_assert(std::is_same<int, typename is_unique_ptr<std::unique_ptr<int>>::underlying_type>::value, "");
 
+template <typename B, typename E>
+struct can_be_stored_in_unique_ptr {
+  static constexpr bool value = false;
+};
+
+// Note: No custom deleters here [yet]. TODO(dkorolev): Make sure they are here as we need them.
+template <typename B, typename E>
+struct can_be_stored_in_unique_ptr<std::unique_ptr<B>, E> {
+  static constexpr bool value = std::is_same<B, E>::value || std::is_base_of<B, E>::value;
```

Comment:
`std::is_base`: If Derived is derived from Base or if both are the same non-union class, provides the member constant value equal to true. Otherwise value is false.
So, `is_same` is redundant.